### DISCUSSION TITLE
Integrating ClearML Debug Samples feature

### DISCUSSION THREAD

**Discussion by hanshupe007** [mindee/doctr#1727] [Ideas]
As ClearML is already being used for logging the training process, leveraging it's visualization feature ("Debug Samples") is very useful to investigate the incorrectly predicted test samples for each epoch.

This can easily be implemented with the report_image function. The screenshot below shows the result. Similar it can be done for the detection model training.

`Logger.current_logger().report_image(title="Recognition Errors", series=title, iteration=int(epoch), image=output)
`

![image](https://github.com/user-attachments/assets/ccc72cc6-62d7-49ec-8224-ac08f3467731)

**Comment by felixdittrich92** [comment]
Hi @hanshupe007 :wave:,

Thanks for the showcase looks really useful :+1: 

As already mentioned #1458 is still assigned to @odulcy-mindee .
But as i see you work already with ClearML would you be interested to add ClearML support to each of the tensorflow + pytorch training scripts ? (Always happy about new contibutors :hugs:)

In this case we could finish one of the two tasks / Slack logging would be still depend on Olivier :)
